src/components/pill/pill.style.js

```diff
@@ -16,12 +19,24 @@ function addStyleToPillIcon(fontSize) {
 
 const PillStyle = styled.span`
  ${({
-    colorVariant, theme, inFill, isDeletable, pillRole, size, ml, mr
+    colorVariant, borderColor, theme, inFill, isDeletable, pillRole, size, ml, mr
   }) => {
     const isStatus = pillRole === 'status';
     const { colors, text } = baseTheme;
     const variety = isStatus ? colorVariant : 'primary';
-    const { varietyColor, buttonFocus } = styleConfig(theme)[pillRole][variety];
+    let pillColor;
+    let buttonFocusColor;
+
+    if (borderColor) {
+      pillColor = toColor(theme, borderColor);
+      buttonFocusColor = shade(0.2, pillColor);
+    } else {
+      const { varietyColor, buttonFocus } = styleConfig(theme)[pillRole][variety];
+      pillColor = varietyColor;
+      buttonFocusColor = buttonFocus;
+    }
+
+    const contentColor = meetsContrastGuidelines(pillColor, text.color).AAA ? text.color : colors.white;
```

Comment:
Should this feed an error to the user to tell them why their colour was changed to white?

Reply:
Do we even need this in the code? Should really be down to the designer or dev using the component to make sure their colours are accessible

Reply:
`contentColor` = color of the text in the `Pill`
It has to be automated because user can control the color of the Pill but not the color of the pill text